File: Android/app/src/main/java/com/edu/primary/database/entity/QuestionEntity.java
package com.edu.primary.database.entity;

import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.TypeConverters;
import com.edu.primary.database.converter.ListConverter;

import java.util.List;

@Entity(tableName = "questions")
@TypeConverters(ListConverter.class)
public class QuestionEntity {
    @PrimaryKey(autoGenerate = true)
    public long id;

    public int subjectId;
    public int grade;
    public String type; // single_choice, multiple_choice, fill_blank, judgment
    public String content;
    public List<String> options;
    public String correctAnswer;
    public String explanation;
    public int difficulty; // 1-5

    public QuestionEntity() {
    }
}

File: Android-1/app/src/main/java/com/edu/primary/data/local/entity/QuestionEntity.kt
package com.edu.primary.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import com.edu.primary.data.local.converter.StringListConverter

@Entity(tableName = "questions")
@TypeConverters(StringListConverter::class)
data class QuestionEntity(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val subjectId: Int,
    val grade: Int,
    val type: String,
    val content: String,
    val options: List<String>? = null,
    val correctAnswer: String,
    val explanation: String? = null,
    val difficulty: Int = 1
)
